geotrek/core/management/commands/loadpaths.py

```diff
@@ -0,0 +1,92 @@
+                            help="File's SRID")
+        parser.add_argument('--intersect', '-i', action='store_true', dest='intersect', default=False,
+                            help="Check paths intersect spatial extent and not only within")
+
+    def handle(self, *args, **options):
+        verbosity = options.get('verbosity')
+        encoding = options.get('encoding')
+        filename = options.get('paths')
+        structure = options.get('structure')
+        name = options.get('name')
+        if name:
+            name = name.encode(encoding)
+        srid = options.get('srid')
+        do_intersect = options.get('intersect')
+        comments = options.get('comments')
+        if comments:
+            comments = comments.encode(encoding)
+
+        if structure or Structure.objects.count() > 1:
+            try:
+                structure = Structure.objects.get(name=structure)
+            except Structure.DoesNotExist:
+                raise CommandError("Structure does not match with instance's structures\n"
+                                   "Use --structure to define it or change your option structure")
+        else:
+            structure = Structure.objects.first()
+        if verbosity > 0:
+            self.stdout.write("All paths in DataSource will be linked to the structure : %s" % structure)
+
+        ds = DataSource(filename)
+
+        bbox = Polygon.from_bbox(settings.SPATIAL_EXTENT)
+        bbox.srid = settings.SRID
+
+        for layer in ds:
+            for feat in layer:
+                try:
+                    geom = feat.geom.geos
+                    if not isinstance(geom, LineString):
+                        if verbosity > 1:
+                            self.stdout.write("%s's geometry is not a Linestring" % feat)
+                        break
+                    self.check_srid(srid, geom)
+                    geom.dim = 2
+                    if do_intersect:
+                        if bbox.intersects(geom):
+                            path = Path.objects.create(name=feat.get(name),
+                                                       structure=structure,
+                                                       geom=geom,
+                                                       comments=comments)
+                            if verbosity > 1:
+                                self.stdout.write('Create path : {}'.format(path.name))
+                    else:
+                        if geom.within(bbox):
+                            path = Path.objects.create(name=feat.get(name),
+                                                       structure=structure,
+                                                       geom=geom,
+                                                       comments=comments)
+                            if verbosity > 1:
+                                self.stdout.write('Create path : {}'.format(path.name))
```

Comment:
Il y a moyen de ne pas dupliquer de code du tout en faisant : `if do_intersect and bbox.intersects(geom) or not do_intersect and geom.within(bbox):`. Comme les opérateurs logiques sont paresseux en python, un seul des deux tests sur la bbox sera effectué.

Reply:
done

Reply:
Done